code/game/objects/items/circuitboards/machine_circuitboards.dm

```diff
@@ -1040,6 +1040,26 @@
 	build_path = /obj/machinery/harvester
 	req_components = list(/obj/item/stock_parts/micro_laser = 4)
 
+/obj/item/circuitboard/machine/medical_kiosk
+	name = "Medical Kiosk (Machine Board)"
+	build_path = /obj/machinery/medical_kiosk
+	var/custom_cost = 10
+	req_components = list(
+		/obj/item/healthanalyzer = 1,
+		/obj/item/stock_parts/scanning_module = 1)
+
+/obj/item/circuitboard/machine/medical_kiosk/multitool_act(mob/living/user)
+	. = ..()
+	var/new_cost = input("Set a new cost for using this medical kiosk.","New cost", custom_cost) as num|null
```

Comment:
I would honestly make this a UI button on the kiosk itself rather than this. Have a "medical authorization" mode which lets you access this.